### PR TITLE
Typescript support for Search

### LINKED ISSUE #569: Refactor Search to Typescript
Create typings for [Search](https://eds.equinor.com/components/search/)

Make sure you fork the repository and branch from `typescript` before starting on this issue

### PR DESCRIPTION
**Pull request by nicholasdalhaug**
Resolves #569

### CODE REVIEW

**Comment by nicholasdalhaug** [comment]
Peer-programmed with @mrSundvoll

**Comment by nicholasdalhaug** [comment]
Note that we have removed redundant properties. We are not sure if the comments over them are necessary as documentation for the storybook. This is because we were not able to view the inputs part of the docs page in the storybook. If these comments are necessary, we can easily add them into Props.

File: libraries/core-react/src/Search/Search.tsx [hunk 1]
```diff
@@ -121,6 +124,10 @@ const Input = styled.input`
     `}
 `
 
+interface InsideButtonProps {
```
Comment (mimarz) [inline libraries/core-react/src/Search/Search.tsx:127]: I assume this should be `type`? 😄

File: libraries/core-react/src/Search/Search.tsx [hunk 2]
```diff
@@ -29,6 +27,11 @@ const {
   },
 } = tokens
 
+type ContainerProps = {
+  isFocused: boolean
+  disabled: boolean
+}
+
 const Container = styled.span`
```
Comment (mimarz) [inline libraries/core-react/src/Search/Search.tsx:35]: This works, but we have used the "[generic](https://styled-components.com/docs/api#using-custom-props)" approach suggested by `styled-components` when applying props to styled components.

Please update so it conforms with the approach we have used in our other components 😅

File: libraries/core-react/src/Search/Search.tsx [hunk 3]
```diff
@@ -121,6 +124,10 @@ const Input = styled.input`
     `}
 `
 
+interface InsideButtonProps {
+  isActive: boolean
+}
+
 const InsideButton = styled.div`
```
Comment (mimarz) [inline libraries/core-react/src/Search/Search.tsx:131]: Same as previous "generic" type comment 🙂

**Review by mimarz — CHANGES_REQUESTED** [review]
Looks mostly good! Just a few clean code comments. 🤞

File: libraries/core-react/src/Search/Search.tsx (hunk 2 above)
Comment (nicholasdalhaug) [reply libraries/core-react/src/Search/Search.tsx]: Done. W did not see that convention. Thanks!

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Comment by nicholasdalhaug** [comment]
Could you label with 'hacktoberfest-accepted'? 😄

**Comment by wenche** [comment]
> Could you label with 'hacktoberfest-accepted'? 😄

Are you sure that's necassary? We read 
```
PRs count if:
Submitted during the month of October AND
Submitted in a public repo AND (
  The PR is labelled as hacktoberfest-accepted by a maintainer OR
  Submitted in a repo with the hacktoberfest topic AND (
    The PR is merged OR
    The PR has been approved
  )
)
``` and it seems like it's covered by the second OR, so we didn't labeled the others PRs we received. :) But ofc, if you feel that is necessary we'll do it @nicholasdalhaug

**Review by wenche — APPROVED** [review]
Thanks a lot for the contribution! 👍

**Comment by mimarz** [comment]
Thanks for the contribution! @nicholasdalhaug @mrSundvoll

**Comment by nicholasdalhaug** [comment]
Thanks! @wenche, I believe it is necessary. I cannot see the hacktoberfest topic on this repo. Also, 
image
and 
image

**Pull request merged**